storage/grpc_client.go

```diff
@@ -136,6 +152,9 @@ func newGRPCStorageClient(ctx context.Context, opts ...storageOption) (storageCl
 }
 
 func (c *grpcStorageClient) Close() error {
+	if c.settings.meterProvider != nil {
```

Comment:
Is it problematic if this never gets called? e.g. will some data not be flushed?

I just ask because calling client.Close() is basically optional; the program could just terminate first. But maybe that's fine.